lib/tictactoe/Bot.ex

```diff
@@ -1,18 +1,18 @@
-defmodule Tictactoe.Bot do
-  @callback calculate_move(Map) :: integer
+defmodule Bot do
+  def calculate_move(:PlayerWon, _, _, _, _), do: nil
+  def calculate_move(:Draw, _, _, _, _), do: nil
 
-  def calculate_move(game_round) do
-      player = :PLAYER_1
-      bot = :PLAYER_2
-      board = game_round.board
-      space = Enum.find_index(
-        board, fn s ->
-          s != player &&
-          s != bot end)
+  def calculate_move(:Playing, board, _, _, :BOT_EASY) do
+    Board.available_spaces(board)
+    |> BotIntellect.random_move
+  end
 
-      %{
-        space: space,
-        player: bot
-      }
+  def calculate_move(:Playing, board, opponent_player, bot_player, :BOT_HARD) do
+    available_spaces = Board.available_spaces(board)
+    move = BotIntellect.take_center(available_spaces)
```

Comment:
These can pipe I think?
```
board
|> Board.availanle_spaces()
|> BotIntellect.take_center()
|> Botintellect.get_winning_move(available_spaces, bot_player, board)
|> BotIntellect.get_winning_move(available_spaces, opponent_player, board)
|> BotIntellect.random_move(available_spaces)
```

Reply:
Improved to this one
```
    available_spaces = Board.available_spaces(board)
    available_spaces
    |> BotIntellect.take_center
    |> BotIntellect.get_winning_move(available_spaces, bot_player, board)
    |> BotIntellect.get_winning_move(available_spaces, opponent_player, board)
    |> BotIntellect.random_move(available_spaces)
```
Don't know how to deal with multiple returns in pipeline. Like here, when we have `available_spaces` and `move`